tests/test_builder.py

```diff
@@ -82,6 +184,90 @@ def test_init_args_entry(self):
         for argName, (argInfo, *_) in EXPERIMENT_INFO["arginfo"].items():
             self.mocked_entries[argInfo.pop("ty")].assert_any_call(argName, argInfo)
 
+    def test_init_sched_opts_entry(self):
+        builder.BuilderApp(
+            name="name",
+            experimentPath="experimentPath",
+            experimentClsName="experimentClsName",
+            experimentInfo=copy.deepcopy(EMPTY_EXPERIMENT_INFO)
+        )
+        pipelineInfo = {
+            "default": "main"
+        }
+        priorityInfo = {
+            "default": 1,
+            "unit": "",
+            "scale": 1,
+            "step": 1,
+            "min": 1,
+            "max": 10,
+            "ndecimals": 0,
+            "type": "int"
+        }
+        self.mocked_entries["StringValue"].assert_any_call("pipeline", pipelineInfo)
+        self.mocked_entries["NumberValue"].assert_any_call("priority", priorityInfo)
+        self.mocked_entries["DateTimeValue"].assert_any_call("timed")
+
+    def test_arguments_from_list_widget(self):
+        listWidget = QListWidget()
+        for name, value in (
+            ("name1", "value1"),
+            ("name2", None),
+        ):
+            widget = QWidget()
+            widget.name = name
+            widget.value = mock.MagicMock(return_value=value)
+            item = QListWidgetItem(listWidget)
+            listWidget.addItem(item)
+            listWidget.setItemWidget(item, widget)
+        app = builder.BuilderApp(
+            name="name",
+            experimentPath="experimentPath",
+            experimentClsName="experimentClsName",
+            experimentInfo=copy.deepcopy(EMPTY_EXPERIMENT_INFO)
+        )
+        args = app.argumentsFromListWidget(listWidget)
+        self.assertEqual(args["name1"], "value1")
+        self.assertNotIn("name2", args)
```

Comment:
Hmm, why does `argumentsFromListWidget()` exclude `None` values from the returned dictionary? I don't remember the context.
I see the `_DateTimeEntry` may return `None`. Is it right to exclude the return value?
If so, please document this (`None` values are not included in the returned dictionary) in `argumentsFromListWidget()`.

Reply:
> why does argumentsFromListWidget() exclude None values from the returned dictionary? I don't remember the context.

This is because I want to use the default value of the parameter in the proxy server.
For example, if `_DateTimeEntry` returns `None`, the submission query doesn't contain the `timed` argument, and according to [ARTIQ-PROXY](https://github.com/snu-quiqcl/artiq-proxy/blob/develop/main.py#L117), it will be set to `None`.

In fact, it is okay to send the `None` value explicitly, and it is just my coding style😅

Reply:
In my opinion, that looks little bit weird because in `_DateTimeEntry`, `None` means that the checkbox is not checked, which is a valid and intended state.
If we remove this and rely on the default behavior of the proxy server, then we are ignoring user's intension.
Of course in this case, the default value is `None` so it would not make any problem, but this seems logically awkward.
Consider the case where the default behavior of the proxy is not equivalent to the `None` value after some updates (although I cannot imagine it, conceptually it is possible).

Therefore, I think we should pass `None` explicitly unless it has some downsides.

Reply:
Umm.. I think it's better to hear you say. I will apply it.